scripts/lib/CIME/build.py

```diff
@@ -258,6 +264,9 @@ def _build_libraries(case, exeroot, sharedpath, caseroot, cimeroot, libroot, lid
             full_lib_path = os.path.join(sharedlibroot, sharedpath)
         elif lib == "mpi-serial":
             full_lib_path = os.path.join(sharedlibroot, sharedpath, "mct", lib)
+        elif lib == "cprnc":
+            full_lib_path = os.path.join(sharedlibroot, compiler, "cprnc")
+            case.set_value("CCSM_CPRNC", os.path.join(full_lib_path))
```

Comment:
You need an extra 'cprnc' at the end of this path: right now it points to the directory, not the executable.

Reply:
I'm surprised wesley's testing didn't fail.

Reply:
@billsacks I think you are misinterpreting this  the directory is sharedlibroot/compiler/ and cprnc is the executable.

Reply:
No, I tested it:

```
$ ./xmlquery -p CPRNC
	CCSM_CPRNC: /Users/sacks/projects/scratch/SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default.20200225_113043_ioba26/bld/gnu/cprnc

$ ls -ld /Users/sacks/projects/scratch/SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default.20200225_113043_ioba26/bld/gnu/cprnc
drwxr-xr-x  18 sacks  1202617992  576 Feb 25 11:32 /Users/sacks/projects/scratch/SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default.20200225_113043_ioba26/bld/gnu/cprnc/

$ ls -ld /Users/sacks/projects/scratch/SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default.20200225_113043_ioba26/bld/gnu/cprnc/cprnc
-rwxr-xr-x  1 sacks  1202617992  114032 Feb 25 11:32 /Users/sacks/projects/scratch/SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default.20200225_113043_ioba26/bld/gnu/cprnc/cprnc*
```

Reply:
I see, sorry.

Reply:
Is this machine specific also? I don't see how our internal testing could have passed if CCSM_CPRNC were not set correctly. @WesCoomber , can you confirm?

Reply:
@billsacks You are correct that build.py pointed to the directory instead of the executable. I have implemented the fix and I added a test to scripts_regression_tests.py to catch cases like this in the future. In this commit:
2b882e4d6df38903d835263ab7514f1fd2ed3d75

Reply:
Thanks!